src/public/orders/sales-order-transaction.ts

```diff
@@ -0,0 +1,65 @@
+import { Transaction } from "../common";
+import { SalesOrderStatus } from "./enums";
+import { SalesOrderCustomFieldMappingPOJO } from "./sales-order-time-range";
+
+/**
+ * Session data that may include auth data set by a data driven auth processes.
+ * May include arbitrary data set by the developer.
+ * Must be JSON serializable. Any method may update the session data,
+ * such as renewing a session token or updating a timestamp.
+ */
+export type SalesOrderSession<T extends object = object> = T & {
+  /**
+  * Auth object set by a data driven auth processes.
+  */
+  readonly auth?: {
+    /**
+     * Populated by the data driven auth processes for basic auth integrations.
+     */
+    readonly username?: string;
+
+    /**
+     * Populated by the data driven auth processes for basic auth integrations.
+     */
+    readonly password?: string;
+
+    /**
+     * Populated by the data driven auth processes for OAuth integrations.
+     */
+    readonly accessToken?: string;
+
+    /**
+     * Populated by the data driven auth processes for API Key integrations.
+     */
+    readonly apiKey?: string;
+  }
+}
+
+/**
+ * The ShpEngine Connect passes this object to every method call. It provides information about the
```

Comment:
```suggestion
 * The ShipEngine Connect passes this object to every method call. It provides information about the
```

Reply:
Agree on the second point. Adding now. On the first point, do you have an alternative suggestion? In the long-term we will probably split the SDKs per app type. But, I don't think it is too confusing for someone working on a sales order app to use a `SalesOrderTransaction`.